cypress/integration/page.spec.js

```diff
@@ -1,7 +1,132 @@
-// WRITE TESTS HERE
+/// <reference types="cypress" />
 
 describe('Page', () => {
-  it('should be visitable', () => {
-    cy.visit('/');
+  const apiUrl = 'https://mate.academy/students-api/'
+
+  beforeEach(() => {
+    cy.visit('/')
+  });
+
+  it('user details should be updated after selecting a different user', () => {
+    cy.intercept(`${apiUrl}` + 'users/5', { fixture: 'user5' });
+    cy.intercept(`${apiUrl}` + 'users/6', { fixture: 'user6' });
+
+    cy.get('button')
+      .contains('5')
+      .click();
+
+    cy.getByDataCy('userName')
+      .should('have.text', 'Chelsey Dietrich');
+
+    cy.get('button')
+      .contains('6')
+      .click();
+
+    cy.getByDataCy('userName')
+      .should('have.text', 'Mrs. Dennis Schulist');
+  });
+
+  it('no request to the server after selecting the same user again', () => {
+    cy.intercept(`${apiUrl}` + 'users/5', cy.spy().as('apiCall'));
+
+    cy.get('button')
+      .contains('5')
+      .click()
+      .click();
```

Comment:
Will it fail if I don't implement any caching? Maybe 2 clicks in the row are too fast to send a second request.

Do not need to change the code if it works as expected.

Reply:
fixed